bundles/org.openhab.core.model.core/src/main/java/org/openhab/core/model/core/internal/ModelRepositoryImpl.java

```diff
@@ -203,8 +189,9 @@ public void reloadAllModelsOfType(final String modelType) {
             // Make a copy to avoid ConcurrentModificationException
             List<Resource> resourceListCopy = new ArrayList<>(resourceSet.getResources());
             for (Resource resource : resourceListCopy) {
-                if (resource != null && resource.getURI().lastSegment().contains(".") && resource.isLoaded()) {
-                    if (modelType.equalsIgnoreCase(resource.getURI().fileExtension())) {
+                if (resource.getURI().lastSegment().contains(".") && resource.isLoaded()) {
+                    if (modelType.equalsIgnoreCase(resource.getURI().fileExtension())
+                            && !resource.getURI().lastSegment().startsWith("tmp_")) {
```

Comment:
Combine nested `if`s and consider new condition in method `removeAllModelsOfType()`.